# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/MediaDecoderStateMachine.h
# Commit: b91bd828c657
# Full Hash: b91bd828c657d85120fe02e57410a0b82ec0acd3
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-11-14 21:46:28
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1172394 - Simplify MediaSink somewhat. r=padenot
#   
#   This patches does several minor things:
#   - Moves SetSink (from setSinkid) to automatic coalescing of multiple calls
#     through a Canonical/Mirror setup instead of a manual atomic counter.
# ==============================================================================

diff -r fdf5dd8ff807 -r b91bd828c657 dom/media/MediaDecoderStateMachine.h
--- a/dom/media/MediaDecoderStateMachine.h	Wed Nov 13 22:40:07 2019 +0000
+++ b/dom/media/MediaDecoderStateMachine.h	Wed Nov 13 22:40:05 2019 +0000
@@ -424,7 +424,7 @@
 
   // Always create mediasink which contains an AudioSink or DecodedStream
   // inside.
-  already_AddRefed<MediaSink> CreateMediaSink(bool aOutputCaptured);
+  already_AddRefed<MediaSink> CreateMediaSink();
 
   // Stops the media sink and shut it down.
   // The decoder monitor must be held with exactly one lock count.
@@ -704,6 +704,10 @@
   // upon reaching the end.
   Mirror<bool> mLooping;
 
+  // The device used with SetSink, or nullptr if no explicit device has been
+  // set.
+  Mirror<RefPtr<AudioDeviceInfo>> mSinkDevice;
+
   // Whether all output should be captured into mOutputTracks. While true, the
   // media sink will only play if there are output tracks.
   Mirror<bool> mOutputCaptured;
@@ -729,9 +733,6 @@
   // Used to distinguish whether the audio is producing sound.
   Canonical<bool> mIsAudioDataAudible;
 
-  // Used to count the number of pending requests to set a new sink.
-  Atomic<int> mSetSinkRequestsCount;
-
  public:
   AbstractCanonical<media::TimeIntervals>* CanonicalBuffered() const;
 